rfc/0077-zone-aware-replica-reads.md

```diff
@@ -0,0 +1,183 @@
+# Meta
+
+| Field          | Value                    |
+|----------------|--------------------------|
```

Comment:
I don't see in this doc how the SDK exposes letting the user set which availability zone / server group the appserver is closest to?

I proposed "server group affinity" naming for this concept, as the feature is, at its most general, about sending to a particular server group rather than a local availability zone.
But I'll happily accept an argument that is too broad and that we should focus the feature and the naming on the first intended use-case.

Reply:
I think the only way would be to ask the user to specify the name of the server group, just like it would appear in the configuration. I'm good with this name.

```
var options = new ClusterOptions();
options.serverGroupAffinity("Group 1");
```

We may even emit some warning if group is not in the config, or leaves the cluster.

Reply:
Yes that's it exactly.  The first version of this feature is expected to require the user to manually specify the server group.  Maybe in future we can somehow interrogate the environment to find out we're in AWS AZ1, and somehow know to map that to Server Group 1.

> We may even emit some warning if group is not in the config, or leaves the cluster.

Yes, maybe a good idea, as long as we're not spammy.  (Which I could see it being with 7.6, with the rate that pushes out configs)

Reply:
I have added section for that. Lets leave the warning part as implementation detail.